lightning/src/ln/channel.rs

```diff
@@ -4114,6 +4127,26 @@ impl<Signer: Sign> Channel<Signer> {
 		}
 	}
 
+	/// Get forwarding information for the counterparty.
+	pub fn get_counterparty_forwarding_info(&self) -> Option<CounterpartyForwardingInfo> {
+		self.counterparty_forwarding_info.clone()
+	}
+
+	pub fn channel_update(&mut self, msg: &msgs::ChannelUpdate) -> Result<(), ChannelError> {
+		let usable_channel_value_msat = (self.channel_value_satoshis - self.counterparty_selected_channel_reserve_satoshis) * 1000;
+		if msg.contents.htlc_minimum_msat >  usable_channel_value_msat {
+			return Err(ChannelError::Close("Minimum htlc value is greater than channel value".to_string()));
+		}
+		self.holder_htlc_minimum_msat = msg.contents.htlc_minimum_msat;
```

Comment:
We shouldnt update this - we set our own htlc_minimum, if the counterparty sets a higher one when relaying to us, that's fine, but they can also decrease it later. Either way, they must not violate our original htlc_minimum.